core/src/main/java/org/elasticsearch/index/shard/StoreRecovery.java

```diff
@@ -200,15 +253,13 @@ private void internalRecoverFromStore(IndexShard indexShard, boolean indexShould
                         // it exists on the directory, but shouldn't exist on the FS, its a leftover (possibly dangling)
                         // its a "new index create" API, we have to do something, so better to clean it than use same data
                         logger.trace("cleaning existing shard, shouldn't exists");
-                        IndexWriter writer = new IndexWriter(store.directory(), new IndexWriterConfig(Lucene.STANDARD_ANALYZER).setOpenMode(IndexWriterConfig.OpenMode.CREATE));
-                        writer.close();
+
```

Comment:
I'm glad you removed this scariness ... does this (dangling index) ever happen?  Aren't we more careful about picking unique directories?  Can we throw `AssertionError` if we get here?  Or at least remove the logging line :)